package_bag/tests.py

```diff
@@ -125,36 +119,94 @@ def test_discover_bags(self):
         mock_rights.return_value.json.return_value = self.rights_service_response
-        assign_rights = RightsAssigner().run()
-        mock_rights.assert_called_with(
-            'http://aquila-web:8000/rights',
-            json={'identifiers': RIGHTS_ID, 'start_date': None, 'end_date': END_DATE})
-        self.assertIsNot(False, assign_rights)
+        for bag in Bag.objects.filter(process_status=Bag.DISCOVERED):
+            assign_rights = RightsAssigner().run()
+            mock_rights.assert_called_with(
+                'http://aquila-web:8000/rights',
+                json={'identifiers': RIGHTS_ID, 'start_date': None, 'end_date': END_DATE})
+            self.assertIsNot(False, assign_rights)
         self.assertEqual(
-            mock_rights.call_count, self.expected_count,
+            mock_rights.call_count, 3,
             "Incorrect number of calls to rights service.")
         for obj in Bag.objects.all():
             self.assertEqual(
                 obj.rights_data, self.rights_service_response["rights_statements"],
                 "Rights JSON was not correctly added to bag in database.")
 
-    def test_create_package(self):
+    def test_serialize_json(self):
+        """Ensures that valid JSON is created"""
+        copy_binaries(VALID_BAG_FIXTURE_DIR, settings.TMP_DIR)
+        for bag in Bag.objects.filter(rights_data__isnull=False):
+            package_root = join(settings.DEST_DIR, bag.bag_identifier)
+            PackageMaker().serialize_json(bag, package_root)
+            with open("{}.json".format(join(package_root, bag.bag_identifier)), "r") as f:
+                serialized = json.load(f)
+            self.assertTrue(is_valid(serialized, "{}_bag".format(bag.origin)))
+
+    def tearDown(self):
+        for d in [settings.TMP_DIR, settings.SRC_DIR, settings.DEST_DIR]:
+            if isdir(d):
+                shutil.rmtree(d)
+
+
+class TestPackageMaker(TestCase):
+    fixtures = ["make_package.json"]
+
+    def setUp(self):
+        set_up_directories([settings.TMP_DIR, settings.SRC_DIR, settings.DEST_DIR])
+
+    def test_run(self):
         """Ensures that packages are correctly created."""
-        add_bags_to_db(settings.TMP_DIR, self.expected_count, rights_data=self.rights_json, process_status=Bag.ASSIGNED_RIGHTS)
         copy_binaries(VALID_BAG_FIXTURE_DIR, settings.SRC_DIR)
-        create_package = PackageMaker().run()
-        self.assertIsNot(False, create_package)
+        for bag in Bag.objects.filter(process_status=Bag.ASSIGNED_RIGHTS):
+            create_package = PackageMaker().run()
+            self.assertIsNot(False, create_package)
+            self.assertEqual(
+                len(listdir(settings.TMP_DIR)), 0,
+                "Temporary directory is not empty.")
         self.assertEqual(
-            len(listdir(settings.TMP_DIR)), 0,
-            "Temporary directory is not empty.")
+            len(listdir(settings.DEST_DIR)), 3,
```

Comment:
I'm seeing you assert on the number 3 a lot - wondering if there's a way you can set a variable for this (either globally or in each test) so it's a bit more obvious what this is asserting against?